From e4b7f3cc43e701891ec635fedb462a442f4b8e7d Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Sat, 18 May 2013 09:04:44 +0200
Subject: [PATCH] Remove static libraries

Static libraries fail to build, we also don't want static libraries
so we remove the targets.
---
 Makefile |   18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index cf77b78..e9e5004 100644
--- a/Makefile
+++ b/Makefile
@@ -16,22 +16,18 @@ OBJECTS = md5.o libimplantisomd5.o checkisomd5.o implantisomd5
 SOURCES = $(patsubst %.o,%.c,$(OBJECTS))
 LDFLAGS += -fPIC
 
-PYOBJS = pyisomd5sum.o libcheckisomd5.a libimplantisomd5.a
+PYOBJS = pyisomd5sum.o libcheckisomd5.o libimplantisomd5.o md5.o
 
-all: implantisomd5 checkisomd5 pyisomd5sum.so libimplantisomd5.a libcheckisomd5.a
+all: implantisomd5 checkisomd5 pyisomd5sum.so
 
 %.o: %.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -O -o $@ $<
 
-implantisomd5: implantisomd5.o libimplantisomd5.a
-	$(CC) $(CPPFLAGS) $(CFLAGS) implantisomd5.o libimplantisomd5.a -lpopt $(LDFLAGS) -o implantisomd5
+implantisomd5: implantisomd5.o libimplantisomd5.o md5.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) implantisomd5.o libimplantisomd5.o md5.o -lpopt $(LDFLAGS) -o implantisomd5
 
-checkisomd5: checkisomd5.o libcheckisomd5.a
-	$(CC) $(CPPFLAGS) $(CFLAGS) checkisomd5.o libcheckisomd5.a -lpopt $(LDFLAGS) -o checkisomd5
-
-libimplantisomd5.a: libimplantisomd5.a(libimplantisomd5.o md5.o)
-
-libcheckisomd5.a: libcheckisomd5.a(libcheckisomd5.o md5.o)
+checkisomd5: checkisomd5.o libcheckisomd5.o md5.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) checkisomd5.o libcheckisomd5.o md5.o -lpopt $(LDFLAGS) -o checkisomd5
 
 pyisomd5sum.so: $(PYOBJS)
 	$(CC) $(CPPFLAGS) $(CFLAGS) -shared -g -fpic $(PYOBJS) $(LDFLAGS) -o pyisomd5sum.so
@@ -55,8 +51,6 @@ install-devel:
 	install -d -m 0755 $(DESTDIR)/usr/$(LIBDIR)
 	install -m 0644 libimplantisomd5.h $(DESTDIR)/usr/include/
 	install -m 0644 libcheckisomd5.h $(DESTDIR)/usr/include/
-	install -m 0644 libimplantisomd5.a $(DESTDIR)/usr/$(LIBDIR)
-	install -m 0644 libcheckisomd5.a $(DESTDIR)/usr/$(LIBDIR)
 
 clean:
 	rm -f *.o *.so *.pyc *.a .depend *~
-- 
1.7.10.4

